src/helpers/checkpointing.py

```diff
@@ -0,0 +1,145 @@
+
+    def create_checkpoints_storage(self, experiment_path):
+        assert os.path.isdir(experiment_path), 'Checkpoint of experiment in folder {} not found.'.format(experiment_path)
+        self._logger.info('Recovering checkpoint information from checkpoint of the experiment stored in {}'.format(experiment_path))
+        return [self.get_checkpoint_structure(checkpoint_file) for checkpoint_file in glob.glob(experiment_path + '*/checkpoint*.yml')]
+
+    def get_checkpoint_structure(self, checkpoint_file):
+        assert os.path.isfile(checkpoint_file)
+
+        def get_local_individuals(individual_type, checkpoint_data):
+             return [individual['source'] for individual in checkpoint_data[individual_type]['individuals'] if individual['is_local']]
+
+        def get_adjacent_individuals(individual_type, checkpoint_data):
+              return [create_cell_info(individual['source']) for individual in checkpoint_data[individual_type]['individuals'] if
+                    not individual['is_local']]
+
+        def get_learning_rate(individual_type, checkpoint_data):
+             return checkpoint_data[individual_type]['learning_rate']
+
+        checkpoint_data = yaml.load(open(checkpoint_file))
+        checkpoint = dict()
+        checkpoint['local_generators'] = get_local_individuals('generators', checkpoint_data)
+        checkpoint['local_discriminators'] = get_local_individuals('discriminators', checkpoint_data)
+        checkpoint['discriminators_learning_rate'] = get_learning_rate('discriminators', checkpoint_data)
+        checkpoint['generators_learning_rate'] = get_learning_rate('generators', checkpoint_data)
+        checkpoint['adjacent_individuals'] = get_adjacent_individuals('generators', checkpoint_data)
+
+        checkpoint['iteration'] = checkpoint_data['iteration']
+        checkpoint['time'] = checkpoint_data['time']
+        checkpoint['id'] = checkpoint_data['id']
+        checkpoint['path'] = dirname(checkpoint_file)
+        checkpoint['grid_size'] = checkpoint_data['grid_size']
+        checkpoint['position'] = (checkpoint_data['position']['x'], checkpoint_data['position']['y'])
+
+        self._logger.info(
+            'Recovered checkpoint information from checkpoint file {}.\nCheckpoint information: {}'.format(checkpoint_file, checkpoint))
+
+        return checkpoint
+
+    def get_population_cell_info(self):
+        client_soueces = self.get_population_sources()
+        clients = []
+        for source in client_soueces:
+            clients.append(create_cell_info(source))
+        return clients
+
+    def get_population_sources(self):
+        sources = set()
+        for checkpoint in self.checkpoints_storage:
+            [sources.add(local_genarator) for local_genarator in checkpoint['local_generators']]
+            [sources.add(local_discriminator) for local_discriminator in checkpoint['local_discriminators']]
+        return list(sources)
+
+    def get_local_generators_paths(self, source):
```

Comment:
Looks like code duplication for 2 methods, that could be simplified with function parameters e.g. `get_checkpoint_local_values(self, source, key)`